source/ai/KeeperAI.cpp

```diff
@@ -757,10 +757,12 @@ bool KeeperAI::handleWorkers()
 
     mCooldownWorkers = Random::Int(3,10);
 
-    Tile* central = getDungeonTemple()->getCentralTile();
-    std::vector<Tile*> tiles;
-    tiles.push_back(central);
-    int summonCost = Spell::getSpellCost(&mGameMap, SpellType::summonWorker, tiles, &mPlayer);
+    // We want to use the first covered tile because the central might be destroyed and enemy claimed
```

Comment:
Could you use the first tile as a fallback? So we keep the central position whenever possible? Looks better IIRC.